streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -475,7 +518,7 @@ private void handleTasksWithoutStateUpdater(final Map<TaskId, Set<TopicPartition
                     tasksToRecycle.put(task, standbyTasksToCreate.get(taskId));
                 }
                 standbyTasksToCreate.remove(taskId);
-            } else {
+            } else if (!recycledInitialTasks.contains(taskId)) {
```

Comment:
Given the current code, ` tasks.allTasks()` should not contain any "initialized tasks" thus it seems we don't need this guard as it would always be true (this might change based on the comment above?)?

Reply:
We add our pending standby tasks to the `tasks` on lines 491 (for active tasks) and 496 (for standbys) above.